### PR TITLE
Refactor SqlQueryManager to get query types without down cast

### PR DESCRIPTION
**Pull request by highker**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java [hunk 1]
```diff
@@ -430,7 +408,8 @@ public QueryInfo createQuery(SessionContext sessionContext, String query)
             }
             if (statement instanceof Explain && ((Explain) statement).isAnalyze()) {
                 Statement innerStatement = ((Explain) statement).getStatement();
-                if (!(executionFactories.get(innerStatement.getClass()) instanceof SqlQueryExecutionFactory)) {
+                QueryType innerQueryType = STATEMENT_QUERY_TYPES.get(innerStatement.getClass());
+                if (innerQueryType == null || innerQueryType == QueryType.DATA_DEFINITION) {
```
Comment (wenleix) [inline presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java]: Maybe change the exception message to be

```
    throw new PrestoException(NOT_SUPPORTED, "EXPLAIN ANALYZE doesn't support statement type:  " + innerStatement.getClass().getSimpleName());
```

But I am not sure...

Comment (nezihyigitbasi) [reply presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java]: I think the message can be updated per @wenleix 's suggestion.

File: presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java [hunk 2]
```diff
@@ -430,8 +408,9 @@ public QueryInfo createQuery(SessionContext sessionContext, String query)
             }
             if (statement instanceof Explain && ((Explain) statement).isAnalyze()) {
                 Statement innerStatement = ((Explain) statement).getStatement();
-                if (!(executionFactories.get(innerStatement.getClass()) instanceof SqlQueryExecutionFactory)) {
-                    throw new PrestoException(NOT_SUPPORTED, "EXPLAIN ANALYZE only supported for statements that are queries");
+                QueryType innerQueryType = STATEMENT_QUERY_TYPES.get(innerStatement.getClass());
+                if (innerQueryType == null || innerQueryType == QueryType.DATA_DEFINITION) {
+                    throw new PrestoException(NOT_SUPPORTED, "EXPLAIN ANALYZE doesn't support statement type:  " + innerStatement.getClass().getSimpleName());
```
Comment (nezihyigitbasi) [inline presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java]: there is an extra space after ":" -> `type:  "`

**Comment by highker** [comment]
Per discussion with @martint, we will use the map and store it in a util class.

**Pull request merged**